src/api/app/helpers/webui/webui_helper.rb

```diff
@@ -355,6 +355,20 @@ def user_with_realname_and_icon(user, opts = {})
     end
   end
 
+  def requester_str(creator, requester_user, requester_group)
+    # we don't need to show the requester if he is the same as the creator
+    return if creator == requester_user
+    if requester_user
+      "the user #{user_with_realname_and_icon(requester_user, no_icon: true)}"
+    elsif requester_group
+      "the group #{requester_group}"
+    end
+  end
+
+  def creator_intentions(is_bugowner, role = nil)
+    is_bugowner ? 'become bugowner' : "get the role #{role}"
```

Comment:
Then do now
```
role.blank? ? 'become bugowner' : "get the role #{role}"
```
and later on
```
 role == :bugowner ? 'become bugowner' : "get the role #{role}"
```
Maybe then you don't even need the helper anymore because you can do ``get the role bugowner``

Anyway, I still don't think it is a good idea to have two parameters which hold (almost) the same data and are dependent.

Reply:
@ChrisBr 

> Maybe then you don't even need the helper anymore because you can do get the role bugowner

No, because there is a get role bugowner and a set bugowner and they are different things. The set bugowner remove previous bugowners.